Build error with sdk NONOSDK3V0

Hi,
After latest commit from @d-a-v i have an error due to static_assert if i change the following line in platform.txt
build.sdk=NONOSDK3V0 instead of build.sdk=NONOSDK221.

The compiler give me this error:
.\esp8266\libraries\ESP8266WiFi\src\ESP8266WiFiSTA.cpp: In function 'bool sta_config_equal(const station_config&, const station_config&)':
.\esp8266\libraries\ESP8266WiFi\src\ESP8266WiFiSTA.cpp:67:5: error: static assertion failed: struct station_config has changed, please update comparison function
     static_assert(sizeof(station_config) == 112, "struct station_config has changed, please update comparison function");

Error does not occur when building with SDK v2.2.1

--- a/libraries/ESP8266WiFi/src/ESP8266WiFiSTA.cpp
+++ b/libraries/ESP8266WiFi/src/ESP8266WiFiSTA.cpp
@@ -64,7 +64,11 @@ static bool sta_config_equal(const station_config& lhs, const station_config& rh
  */
 static bool sta_config_equal(const station_config& lhs, const station_config& rhs) {
 
+#ifdef NONOSDK3V0
+    static_assert(sizeof(station_config) == 116, "struct station_config has changed, please update comparison function");
+#else
     static_assert(sizeof(station_config) == 112, "struct station_config has changed, please update comparison function");
+#endif
 
     if(strncmp(reinterpret_cast<const char*>(lhs.ssid), reinterpret_cast<const char*>(rhs.ssid), sizeof(lhs.ssid)) != 0) {
         return false;

